### PR TITLE
Remove contextChanged event and ICodeAllowList

### PR DESCRIPTION
**Pull request by ChumpChief**
[AB#4999](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/4999) - followup to #16372 and also #14211 since I forgot about ICodeAllowList in #14386.

### CODE REVIEW

File: .changeset/hungry-hotels-call.md
Comment (alexvy86) [inline .changeset/hungry-hotels-call.md:1]: Nit: the changesets can be a single file, right?

**Review by alexvy86 — APPROVED** [review]
LGTM, but I don't have a ton of context in this area so take the approval with however much salt you think that's worth 😄.

File: .changeset/hungry-hotels-call.md
Comment (anthony-murphy) [reply .changeset/hungry-hotels-call.md:1]: i think the guidance is a change set per change, so multiple is the right approach here

Comment (ChumpChief) [reply .changeset/hungry-hotels-call.md:1]: I wasn't really sure what best practices were.  The two changes are unrelated to one another, so I imagined the generated change log might read better without having two seemingly-unrelated updates grouped together?

**Pull request merged**